proxy/src/auth/backend/jwt.rs

```diff
@@ -241,31 +269,62 @@ impl JwkCacheEntryLock {
             key => bail!("unsupported key type {key:?}"),
         };
 
-        // TODO(conrad): verify iss, exp, nbf, etc...
+        let now = SystemTime::now();
```

Comment:
`now` isn't needed until checking payload.exp/.nbf. Consider moving it there unless now needs to be taken really at this point, in which case this should be documented.

Reply:
I considered it, but I feel like it's little benefit for more code complexity.

Reply:
Sorry, I don't understand. Why does moving this line down to 289 make the code more complex?

Reply:
Oh - I thought you would only want to sample the conditionally time _if_ we check exp/nbf. Yeah I can move it down